### PR TITLE
Release workflow updates loosely converted to TypeScript

### PR DESCRIPTION
**Pull request by 3mcd**
This PR contains all the fixes @djagdish introduced in #1023, rebased on latest master, with each affected change "rewritten" in TypeScript.

Test Plan:
- Verify this code works exactly the same as #1023.

### CODE REVIEW

**Review by idreyn — CHANGES_REQUESTED** [review]
Great usability improvements here! My feedback is mostly on cosmetic issues, and this should be good to go after a look at some of this stuff relating to copy, icons, padding, and a few questions about CSS structure.

Not sure if I am addressing this at @3mcd or @djagdish here — in fact, I would be happy to go through and make these changes myself with your combined blessing.

File: client/components/PubReleaseDialog/PubReleaseDialog.tsx [hunk 1]
```diff
@@ -16,7 +28,7 @@ type OwnProps = {
 	isOpen: boolean;
 	pubData: {
 		id?: string;
-		releases?: {}[];
+		releases?: { createdAt?: string }[];
```
[the remark this answers is not in the document]
Comment (3mcd) [reply client/components/PubReleaseDialog/PubReleaseDialog.tsx:31]: Ah, I wasn't aware that's how PropTypes worked. I was under the impression that in the above example, `releases` would still be optional unless you tacked an `isRequired` on the end.

File: client/components/PubReleaseDialog/PubReleaseDialog.tsx [hunk 2]
```diff
@@ -91,20 +105,42 @@ const PubReleaseDialog = (props: Props) => {
...
-					The most recent release
-				</a>{' '}
-				was made {dateString}. This will update the version of the Pub that readers see, but
-				they'll still be able to view previous releases by clicking the{' '}
-				<Icon icon="history" iconSize={12} /> History button on the Pub header.
-			</p>
+			<React.Fragment>
+				<p>
+					<Icon icon="history" iconSize={12} color="#888888" />{' '}
+					{' The previous Release ('}
```
Comment (3mcd) [inline client/components/PubReleaseDialog/PubReleaseDialog.tsx:114]: I think we should be able to omit the bracket/quotations from this line (and line 118).

**Comment by 3mcd** [comment]
@idreyn FWIW you have my go-ahead to make any proposed changes, although I can't speak for @djagdish, since most of the suggestions are related to his code!

File: client/components/PubReleaseDialog/PubReleaseDialog.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (idreyn) [reply client/components/PubReleaseDialog/PubReleaseDialog.tsx:31]: Honestly I think you're right, and my mental model of PropTypes is broken!

**Comment by 3mcd** [comment]
I went ahead and knocked out everything but your $0.02 items for now, @idreyn!

File: client/components/PubReleaseDialog/PubReleaseDialog.tsx [hunk 3]
```diff
@@ -126,10 +215,12 @@ const PubReleaseDialog = (props: Props) => {
 		return (
 			<React.Fragment>
 				<Button disabled={isCreatingRelease} onClick={onClose}>
-					Cancel
+					Return to Draft
```
[the remark this answers is not in the document]
Comment (3mcd) [reply client/components/PubReleaseDialog/PubReleaseDialog.tsx:216]: I'm gonna consider this a $0.02 item for now.

**Pull request merged**